### DISCUSSION TITLE
How to configure automx

### DISCUSSION THREAD

**Discussion by vsv87** [modoboa/modoboa#2927] [General]
Hi!

How to properly configure automx on modoboa 2.0.5? It doesn't  work.

**Comment by phpony** [comment]
There's a lot of problems with outdated automx script not working with modern memcached and mysql/psql backends.

I've found an ultimate solution working on all of my servers. I just switch it to static mode - no backends = no problems! 

Just set /etc/automx.conf to this:

```
[automx]
provider = mail.your.domain
domains = *

[global]
backend = static
action = settings
account_type = email

smtp = yes
smtp_server = mail.your.domain
smtp_port = 587
smtp_encryption = starttls
smtp_auth = plaintext
smtp_auth_identity = %s
smtp_refresh_ttl = 6
smtp_default = yes

imap = yes
imap_server = mail.your.domain
imap_port = 143
imap_encryption = starttls
imap_auth = plaintext
imap_auth_identity = %s
imap_refresh_ttl = 6

#pop = yes
#pop_server = mail.your.domain
#pop_port = 110
#pop_encryption = starttls
#pop_auth = plaintext
#pop_auth_identity = %s
```

Save, restart uwsgi and - voila! - it's working (in most cases).

**Reply by Spitfireap** [reply]
Thanks, we wanted to migrate to automx2 at some point. Ill test it. Would you be able to create a PR ?

**Reply by metaldigger** [reply]
Damn, i've the same issue, i've made the @phpony configuration with backend = static but when i browse autoconfig.xxxx.com i get "Internal Server Error". The only error i found is in automx_instance.log: "no python application found, check your startup logs for errors".

**Reply by phpony** [reply]
Had similar error on one of the servers. You probably have unresolved dependencies. Try running:

```
# su - automx
# source env/bin/activate
# pip3 install "sqlalchemy < 2.0"
# pip3 install python-ldap 
# pip3 install mysqlclient
# pip3 install M2Crypto 
# pip3 install ipaddr
# pip3 uninstall memcache
```

Note - we're removing memcache, not installing! Then test the script itself is working:

`# python3 instance/automx_wsgi.py`

If there's no error - restart uwsgi as root and then run as automx to check uwsgi is serving it right:

`# automx-test example@xxxx.com`

**Reply by metaldigger** [reply]
Hi @phpony, thank you! You're right, on the server there are sqlalchemy and mysqlclient unresolved dependencies but when i test this script "python3 instance/automx_wsgi.py" i get this error: 

"File automx_wsgi.py line 50 in <module> 
from automx.config import Config
ModuleNotFioundError: No Module named 'automx' "

**Reply by phpony** [reply]
Well, it's `pip3 install automx` time then)

**Reply by metaldigger** [reply]
I tried but with `pip3 install automx` i get 
ERROR: Could not find a version that satisfied the requirement automx (from versions: none)
ERROR: No matching distribution found for automx

**Reply by phpony** [reply]
A version of automx should've been shipped with Modoboa already. You can find it in "/srv/automx/master.zip" archive. This zip is probably already extracted to /srv/automx/automx-master folder if installation went right. Just copy the module code to your python site-packages folder as automx user:

```
# su - automx
# cp -R automx-master/src/automx env/lib/python3.9/site-packages/
```

Then run `pip3 show automx` to check it. Mind your python version there. Mine is 3.9 but yours can be different.

**Reply by phpony** [reply]
You can copy it as root user too, just chown the folder to automx user afterwards.

![image](https://github.com/modoboa/modoboa/assets/480287/0048b52a-ed3f-4808-a770-2759e03d31a7)

```
$ pip3 show automx
Name: automx
Version: 1.1.1
Summary: Provides account configuration data to mailclients
Home-page: https://github.com/sys4/automx
Author: Christian Roessner
Author-email: c@roessner.co
License: GNU GPLv3
Location: /srv/automx/env/lib/python3.9/site-packages
Requires: 
Required-by:
```

**Reply by metaldigger** [reply]
My python version is 3.10 so i've /srv/automx/env/lib/python3.10/site-packages/automx but when i run `pip3 show automx` i get 
`WARNING: Package(s) not found: automx`

**Reply by xinomilo** [reply]
same with d12 and python3.11. seems automx is deprecated by now (?)
other options? looking into automx2 and email-autoconf ... not sure which one is suited best for modoboa..
